opencog/query/PatternUtils.cc

```diff
@@ -117,21 +116,18 @@ void get_connected_components(
 			size_t nc = components.size();
 			for (size_t i = 0; i<nc; i++)
 			{
-				std::set<Handle> cur_vars(component_vars[i]);
```

Comment:
I did check very carefully at the time. Actually I think it would benefit the code readability and safety to consity all that code. For instance node_match arguments could be const, the fact that they are not const suggests to the reader (to me at least) that they might be changed while it's not the case. I'll probably create a branch for that as well.